cloud-storage-reference-app.md

```diff
@@ -0,0 +1,366 @@
+
+The CHCS service is a temporary solution being utilized for the Preliminary Blood Pressure Prototype project.
+
+# Problem
+
+The cryptographic requirements for healthcare providers and researchers to access and integrate with CHCS are cumbersome and impeding adoption. In addition given that the technology stacks for the Agile Metabolic Health project will be deployed within the healthcare organization HIPAA compliant environment the added cyrptographic requirements are not necessary.
+
+# Proposal
+
+Drop the requirement to host CHCS on third-party servers and therefore the dependency on cryptography and instead support healthcare industry standards for authentication, authorization and integration.
+
+The new Cloud Storage Service implementation will support the following use cases for researcher end-users:
+
+1. Provider and Researcher access:
+   1. Administrator sets up organization hierarchies for institution, department, etc. (Not displayed in diagram)
+   1. Administrator creates study groups and invites Providers and Researchers to study groups.
+   1. Provider or Researcher signs-in from web browser or from a SMART on FHIR EHR launch.
+1. Provider searches for existing patients or registers new patients. If SMART on FHIR launch, patient details are pulled from the EHR context.
+1. Provider invites selected patients to selected study group (pending consent).
+1. Deep Link is sent by SMS or E-mail to invited patients.
+1. Patient launches deep link via Common Health Android app and gives explicit consent for their specific personal device data to be shared with a specific study group.
+1. The Patient authorizes the Common Health Android app to sync with the personal device vendor's api.
+1. The Common Health Android app periodically downloads vendor personal device data to the patient's Android device.
+1. The Common Health Android app periodically uploads pesonal device data in Open mHealth format to the Cloud Storage Service.
+1. Jupyter Health hub programmatically accesses the Cloud Storage Service REST API with OAuth2/OIDC.
+1. Researcher interacts with study group data via a Jupyter Health hub creating intuitive dashboards for the Provider to view and interact with the study group data.
+1. Provider interacts with the Jupyter Health dashboards. Initially this may be a simple web application, but will eventually be a Provider SMART app within the EHR vendor environment.
+1. A REST API with OAuth2 for programmatic access
+
+![](assets/images/chcs_iam_overview.jpg)
```

Comment:
The submitted image is named `css_iam_overview.jpg` rather than `chcs_iam_overview.jpg`.

Reply:
Fixed